src/rosdiscover/ros2_translator/arch.py

```diff
@@ -0,0 +1,41 @@
+import yaml
+from launch.actions import IncludeLaunchDescription
+from launch_ros.actions import Node
+from launch.launch_description_sources import get_launch_description_from_python_launch_file
+
+description = get_launch_description_from_python_launch_file('robot.launch.py')
+nodes = description.entities
+
+
+def toyaml(nodes):
+    to_yaml = {}
+    no_name = 0
+    for node in nodes:
+        if isinstance(node, IncludeLaunchDescription):
+            description1 = node._IncludeLaunchDescription__launch_description_source.try_get_launch_description_without_context()
+            if description1 is None:
+                continue
+            nodes1 = description1.entities
+            to_yaml.update(toyaml(nodes1))
+        if isinstance(node, Node):
+            dic = {}
+            if node._Node__node_name is not None:
```

Comment:
In general, we shouldn't be accessing private attributes. This is really a hack that should be used in very few situations. I suspect that there are some publicly visible properties, methods, or attributes that we could use to get the same information?

Reply:
Yeah it really is. When I looked, I couldn't find a method except for accessing the node name. I'll look again though.

Reply:
`node.node_name` _should_ work here (https://github.com/ros2/launch_ros/blob/master/launch_ros/launch_ros/actions/node.py#L320)

Reply:
It looks like you're right. There are a few useful getter properties that give us access to some information, but other important parts are made private with `__`. Where we can't use the public API to get access to the information we need, this is OK, but we should maybe add a small comment in the code to justify the decision (and to remind us in the future that this script may break with changes to the implementation of `ros_launch` and `launch`).